### PR TITLE
Crescendo

### PR DESCRIPTION
**Pull request by austinkline**

### CODE REVIEW

File: contracts/standard/ExampleToken.cdc [hunk 1]
```diff
@@ -52,16 +54,36 @@ pub contract ExampleToken: FungibleToken {
...
+    access(all) resource Vault: FungibleToken.Vault {
 
         /// The total balance of this vault
-        pub var balance: UFix64
+        access(all) var balance: UFix64
 
         // initialize the balance at resource creation time
         init(balance: UFix64) {
             self.balance = balance
         }
```
Comment (FelipeRibeiroLabs) [inline contracts/standard/ExampleToken.cdc:66]: We are missing burnCallback fn here:

```
/// Called when a fungible token is burned via the `Burner.burn()` method
        access(contract) fun burnCallback() {
            if self.balance > 0.0 {
                ExampleToken.totalSupply = ExampleToken.totalSupply - self.balance
            }
            self.balance = 0.0
        }```

File: contracts/LostAndFound.cdc [hunk 2]
```diff
@@ -503,45 +513,45 @@ pub contract LostAndFound {
             return false
         }
 
-        pub fun setLowBalanceThreshold(threshold: UFix64?) {
+        access(Mutate) fun setLowBalanceThreshold(threshold: UFix64?) {
```
Comment (web3ashlee) [inline contracts/LostAndFound.cdc:516]: Where does the access(Mutate)entitlement get defined? It's not in L&F contract.

Edit: Never mind this is a built-in entitlement (and combines Insert/Remove entitlements)

File: transactions/example-nft/setup.cdc [hunk 3]
```diff
@@ -3,29 +3,44 @@ import "ExampleNFT"
...
-            target: ExampleNFT.CollectionStoragePath
-        )
+        var pubCap = signer.capabilities.get<&ExampleNFT.Collection>(cd.publicPath)
+        if pubCap == nil {
+            pubCap = signer.capabilities.storage.issue<&ExampleNFT.Collection>(cd.storagePath)
+        } else if !pubCap!.check() {
+            pubCap = signer.capabilities.storage.issue<&ExampleNFT.Collection>(cd.storagePath)
+        }
+
+        // does the public capability exist and succeed?
```
Comment (web3ashlee) [inline transactions/example-nft/setup.cdc:25]: Is this the new unlink/link?

File: transactions/example-nft/try_send_example_nft.cdc [hunk 4]
```diff
@@ -6,49 +6,56 @@ import "MetadataViews"
...
-        self.flowProvider = acct.getCapability<&FlowToken.Vault{FungibleToken.Provider}>(flowTokenProviderPath)
-        self.flowReceiver = acct.getCapability<&FlowToken.Vault{FungibleToken.Receiver}>(/public/flowTokenReceiver)
+        self.flowProvider = provider!
+        self.flowReceiver = sender.capabilities.get<&FlowToken.Vault>(/public/flowTokenReceiver)!
+
+        self.nftReceiverCap = receiver.capabilities.storage.issue<&{NonFungibleToken.Collection}>(ExampleNFT.CollectionStoragePath)
+
+        if revoke {
+            receiver.capabilities.storage.getController(byCapabilityID: self.nftReceiverCap.id)!.delete()
+        }
```
Comment (web3ashlee) [inline transactions/example-nft/try_send_example_nft.cdc:41]: I need to get more familiar with L&F but what dictates if revoke is t/f?  

If revoke is true then we are deleting the capability we created in line 37?

**Review by web3ashlee — APPROVED** [review]
Nice! Most of my comments are questions/thoughts I had while reading through this. Reviewing this was a good way to get more familiar with upgrades. I had to reference [docs](https://cadence-lang.org/docs/1.0/language/accounts/) a few times!

File: contracts/LostAndFound.cdc [hunk 5]
```diff
@@ -344,7 +344,7 @@ pub contract LostAndFound {
         }
 
         // Redeem a specific ticket instead of all of a certain type.
-        pub fun redeem(type: Type, ticketID: UInt64, receiver: Capability) {
+        access(Withdraw) fun redeem(type: Type, ticketID: UInt64, receiver: Capability) {
```
Comment (highskore) [inline contracts/LostAndFound.cdc:347]: if redeemAll is access(all), why does this have to be access(Withdraw)

**Comment by austinkline** [comment]
Leaving a note that I will be revisiting our this PR again since there have been some changes to Crescendo since I did the initial pass-through. I'll flag everyone who has commented once it is ready 🙏

File: transactions/example-nft/try_send_example_nft.cdc (hunk 4 above)
Comment (austinkline) [reply transactions/example-nft/try_send_example_nft.cdc:41]: Correct, this is to test L&F handling a capability that failed its check, essentially

File: contracts/LostAndFound.cdc (hunk 5 above)
Comment (austinkline) [reply contracts/LostAndFound.cdc:347]: Good callout! Fixed

**Pull request merged**